partner_compassion/models/partner_compassion.py

```diff
@@ -517,47 +501,82 @@ def _check_qorder(self, word):
                 raise
         return True
 
+    def _check_duplicates_domain(self, vals=None, skip_props_check=False):
+        if not vals:
+            vals = {
+                "email": self.email,
+                "firstname": self.firstname,
+                "lastname": self.lastname,
+                "zip": self.zip,
+                "street": self.street,
+            }
+
+        # define set of checks for duplicates and required fields
+        checks = [
+            # Email check
+            (vals.get('email'), [
+                '&',
+                ('email', '=', vals.get('email')),
+                ('email', '!=', False)
+            ]),
+            # zip and name check
+            (vals.get('firstname') and vals.get('lastname') and vals.get('zip'), [
+                "&",
+                "&",
+                ("firstname", "ilike", vals.get('firstname')),
+                ("lastname", "ilike", vals.get('lastname')),
+                ("zip", "=", vals.get('zip')),
+            ]),
+            # name and address check
+            (vals.get('lastname') and vals.get('street') and vals.get('zip'), [
+                "&",
+                "&",
+                ("lastname", "ilike", vals.get('lastname')),
+                ("zip", "=", vals.get('zip')),
+                ("street", "ilike", vals.get('street'))
+            ])
+        ]
+
+        # This step builds a domain query based on the checks that
+        # passed, prepending the list with a "|" operator for all item
+        # once its size is > 1
+        search_filters = []
+        for check in checks:
+            if skip_props_check or check[0]:
```

Comment:
Shouldn't be this `if not skip_props_check or check[0]:`, or is the variable name confusing? I'm not sure to understand the use of it, maybe a method comment would be helpful with parameters description.